packages/artifact/src/upload-artifact-http-client.ts

```diff
@@ -0,0 +1,300 @@
+  console.log(
+    `Uploading chunk of size ${end -
+      start +
+      1} bytes at offset ${start} with content range: ${getContentRange(
+      start,
+      end,
+      totalSize
+    )}`
+  )
+
+  const requestOptions = getRequestOptions()
+  requestOptions['Content-Type'] = 'application/octet-stream'
+  requestOptions['Content-Length'] = totalSize
+  requestOptions['Content-Range'] = getContentRange(start, end, totalSize)
+
+  const uploadChunkRequest = async (): Promise<IHttpClientResponse> => {
+    return await restClient.sendStream('PUT', resourceUrl, data, requestOptions)
+  }
+
+  const response = await uploadChunkRequest()
+
+  if (!response.message.statusCode) {
+    console.log(response)
+    throw new Error('No Status Code returned with response')
+  }
+
+  if (isSuccessStatusCode(response.message.statusCode)) {
+    debug(
+      `Chunk for ${start}:${end} was succesfully uploaded to ${resourceUrl}`
+    )
+    return
+  }
+  if (isRetryableStatusCode(response.message.statusCode)) {
+    console.log(
+      `Received ${response.message.statusCode}, will retry chunk at offset ${start} after 10 seconds.`
+    )
+    await new Promise(resolve => setTimeout(resolve, 10000))
+    console.log(`Retrying chunk at offset ${start}`)
+
+    const retryResponse = await uploadChunkRequest()
+    if (!retryResponse.message.statusCode) {
+      console.log(retryResponse)
+      throw new Error('No Status Code returne with response')
+    }
+    if (isSuccessStatusCode(retryResponse.message.statusCode)) {
+      return
+    }
+  }
+}
+
+/**
+ * Step 3 of 3 when uploading an artifact
+ * Updates the size of the artifact from -1 which was initially set during step 1. Updating the size indicates that we are
+ * done uploading all the contents of the artifact. A server side check will be run to check that the artifact size is correct
+ * for billing purposes
+ */
+export async function patchArtifactSize(
+  size: number,
+  artifactName: string
+): Promise<void> {
+  const requestOptions = getRequestOptions()
+  requestOptions['Content-Type'] = 'application/json'
+
+  let resourceUrl = new URL(getArtifactUrl())
+  resourceUrl.searchParams.append('artifactName', artifactName)
+
+  let parameters: PatchArtifactSize = <PatchArtifactSize>{Size: size}
+  let data: string = JSON.stringify(parameters, null, 2)
+
+  const token = process.env['ACTIONS_RUNTIME_TOKEN'] || ''
+  const bearerCredentialHandler = new BearerCredentialHandler(token)
+  let client: HttpClient = new HttpClient('actions/artifact', [
+    bearerCredentialHandler
+  ])
+
+  console.log(`URL is ${resourceUrl.toString()}`)
+
+  let rawResponse: HttpClientResponse = await client.patch(
+    resourceUrl.toString(),
+    data,
+    requestOptions
+  )
+  let body: string = await rawResponse.readBody()
+
+  if (rawResponse.message.statusCode == 200) {
+    let successResponse: PatchArtifactSizeSuccessResponse = JSON.parse(body)
+    console.log('Artifact size was succesfully updated!')
```

Comment:
Although we are patching for size, but to users this is just finishing up uploading artifacts as a whole.  I would say change this line to indicate artifact uploaded successfully.  You could even move the log around artifact size here, to something like `Artifact $name uploaded successfully, total size $size`.   

A general comment is we should check our logging in those files and make sure to read them again from users' perspective.

Reply:
Changed all the comments to make them a bit more friendly from a users perspective